File: engines/RecognizerIO/RecognizerIO/TextToSRG/InputStream.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TextToSRG
{
    public class InputStream
    {
        
        private string Input;
        private int Pos;

        public InputStream(string input)
        {
            this.Input = input;
            this.Pos = 0;
        }
        public string Next()
        {
            return Input[Pos++].ToString();
        }
        public string Peek()
        {
            return Input.Length > Pos ? Input[Pos].ToString() : "";
        }
        public bool Eof()
        {
            return this.Peek() == "";
        }
        public void Croak(string msg)
        {
            throw new Exception(msg);
        }
    }
}

File: osspeak/recognition/inputstream.py
class InputStream:

    def __init__(self, text):
        self.text = text
        self.pos = 0

    def next(self):
        ch = self.text[self.pos]
        self.pos += 1
        return ch

    def peek(self):
        return None if self.eof() else self.text[self.pos]

    def eof(self):
        return self.pos >= len(self.text)

    def croak(self):
        pass

    